cmd/sonobuoy/app/imageversion.go

```diff
@@ -88,11 +88,20 @@ func (c *ConformanceImageVersion) Get(client discovery.ServerVersionInterface) (
 			return "", fmt.Errorf("version %q only has %d segments, need at least 2", version.GitVersion, len(segments))
 		}
 
-		// NOTE: Until the kube-conformance container is pushed upstream we can't
-		// guarantee alignment with exact versioning see https://github.com/heptio/kube-conformance/issues/25
-		// for more details
-		// Use the segments instead of .major and .minor because GKE's .minor is `10+` instead of `10`.
-		return fmt.Sprintf("v%d.%d", segments[0], segments[1]), nil
+		// Temporary logic in place to truncate auto-resolved versions while we
+		// transition to upstream. If < 1.13 return 2 segments due to lag behind
+		// releases. Otherwise return 3. Use the segments instead of .major and
+		// .minor because GKE's .minor is `10+` instead of `10`.
+		if segments[0] == 1 && segments[1] < 13 {
+			return fmt.Sprintf("v%d.%d", segments[0], segments[1]), nil
+		}
+
+		// Not sure that this would be hit but default to adding the last
+		// segment as 0 per convention (upstream + semver).
+		if len(segments) < 3 {
+			return fmt.Sprintf("v%d.%d.%d", segments[0], segments[1], 0), nil
```

Comment:
@dims this should be fine, dont you think? Does the server ever report a version with only 2 segments (e.g. 1.11 instead of 1.11.0)? I didn't see any conformance images tagged that way so I thought it safer just to add in the extra 0, just in case. The existing logic seemed to think 2 segments would/could happen.

Reply:
we will need to ask folks like @hoegaarden @calebamiles @spiffxp as i am not quite sure.

Reply:
The server version reports major.minor, the gitVersion reports the whole thing (e.g. `v.1.13.0`)

Reply:
Yes you're right, I wasn't writing very accurately; the question really is more specifically is w.r.t. the `gitVersion`, since that is where these `segments` come from. I wanted to make sure they would have 3 sections or, if they do only report 2 ever, would it be ok to add the `0`.

Reply:
But the real question is do you care about the "dot" release or only major.minor? I think you'd always want to run the latest release which matches your major.minor.

Reply:
So Heptio _wanted_ to track the releases better than major/minor but because it was a manual process they settled on doing just `major.minor` pretty much. It wasn't from any sort of desire or idea that major/minor was all that mattered.

It seems the most intuitive that you'd want to just run the tests that match your server version; that seems like at the time of release the k8s is stating that those are the requirements for your cluster to be conformant.

It seems like a slightly more awkward requirement to make someone use the test image correlating to the most recent patch release of their major.minor if that image wasn't even out when they got their cluster.

That would also make the sonobuoy logic much more difficult. We'd have to reach out to the server and analyze/ filter the list and try to choose an image.

I just dont see a benefit to adding that in when `major.minor.patch` would work just fine and is intuitive if we have those available.